storefront/app/controllers/spree/store_controller.rb

```diff
@@ -228,7 +228,11 @@ def stored_location
       return unless defined?(after_sign_in_path_for)
       return unless defined?(Devise)
 
-      after_sign_in_path_for(Devise.mappings.keys.first)
+      path = after_sign_in_path_for(Devise.mappings.keys.first)
+
+      store_location(path)
```

Comment:
`after_sign_in_path_for` deletes the stored location from the session, so we need to save  it again after reading it